agent/tags.go

```diff
@@ -78,7 +77,7 @@ type tagFetcher struct {
 func (t *tagFetcher) Fetch(ctx context.Context, l logger.Logger, conf FetchTagsConfig) []string {
 	tags := conf.Tags
 
-	if experiments.IsEnabled(ctx, experiments.KubernetesExec) {
+	if t.k8s != nil {
```

Comment:
Is this line right? the `t.k8s` seems to be set always?

Reply:
Good point, on one hand this is defensive in case we set `t.k8s optionally, on the other YAGNI.

Reply:
I changed it so that it used the same logic as before (only run `t.k8s` to create tags if `kubernetes-exec` is enabled).